cli/packages/cmd/ssh.go

```diff
@@ -0,0 +1,518 @@
+	Use:                   "ssh",
+	DisableFlagsInUseLine: true,
+	Args:                  cobra.NoArgs,
+}
+
+var sshIssueCredentialsCmd = &cobra.Command{
+	Example:               `ssh issue-credentials`,
+	Short:                 "Used to issue SSH credentials against a certificate template",
+	Use:                   "issue-credentials",
+	DisableFlagsInUseLine: true,
+	Args:                  cobra.NoArgs,
+	Run:                   issueCredentials,
+}
+
+var sshSignKeyCmd = &cobra.Command{
+	Example:               `ssh sign-key`,
+	Short:                 "Used to sign a SSH public key against a certificate template",
+	Use:                   "sign-key",
+	DisableFlagsInUseLine: true,
+	Args:                  cobra.NoArgs,
+	Run:                   signKey,
+}
+
+func isValidKeyAlgorithm(algo infisicalSdkUtil.CertKeyAlgorithm) bool {
+	switch algo {
+	case infisicalSdkUtil.RSA2048, infisicalSdkUtil.RSA4096, infisicalSdkUtil.ECDSAP256, infisicalSdkUtil.ECDSAP384:
+		return true
+	default:
+		return false
+	}
+}
+
+func getFileName(algo infisicalSdkUtil.CertKeyAlgorithm) string {
+	switch algo {
+	case infisicalSdkUtil.RSA2048:
+		return "id_rsa_2048"
+	case infisicalSdkUtil.RSA4096:
+		return "id_rsa_4096"
+	case infisicalSdkUtil.ECDSAP256:
+		return "id_ecdsa_p256"
+	default:
+		return "id_ecdsa_p384"
+	}
+}
+
+func isValidCertType(certType infisicalSdkUtil.SshCertType) bool {
+	switch certType {
+	case infisicalSdkUtil.UserCert, infisicalSdkUtil.HostCert:
+		return true
+	default:
+		return false
+	}
+}
+
+func writeToFile(filePath string, content string, perm os.FileMode) error {
+    // Ensure the directory exists
+    dir := filepath.Dir(filePath)
+    if err := os.MkdirAll(dir, 0755); err != nil {
+        return fmt.Errorf("failed to create directory %s: %w", dir, err)
+    }
+
+    // Write the content to the file
+    err := os.WriteFile(filePath, []byte(content), perm)
+    if err != nil {
+        return fmt.Errorf("failed to write to file %s: %w", filePath, err)
+    }
+
+    return nil
+}
+
+func issueCredentials(cmd *cobra.Command, args []string) {
+	var infisicalToken string
+	util.RequireLogin()
+	loggedInUserDetails, err := util.GetCurrentLoggedInUserDetails(true)
+	if err != nil {
+		util.HandleError(err, "Unable to authenticate")
+	}
+
+	if loggedInUserDetails.LoginExpired {
+		util.PrintErrorMessageAndExit("Your login session has expired, please run [infisical login] and try again")
+	}
+	infisicalToken = loggedInUserDetails.UserCredentials.JTWToken
+	
+	projectId, err := cmd.Flags().GetString("projectId")
+	if err != nil {
+		util.HandleError(err, "Unable to parse flag")
+	}
+	if projectId == "" {
+		util.PrintErrorMessageAndExit("You must set the --projectId flag")
+	}
+
+	templateName, err := cmd.Flags().GetString("templateName")
+	if err != nil {
+		util.HandleError(err, "Unable to parse flag")
+	}
+	if templateName == "" {
+		util.PrintErrorMessageAndExit("You must set the --templateName flag")
+	}
```

Comment:
Updated including in SSH PR and Go SDK (in `v0.4.6`) so SSH `/sign` and `/issue` endpoints now rely on `certificateTemplateId` instead of `projectId` and `templateName`.